go/cmd/dolt/commands/clone.go

```diff
@@ -301,58 +301,80 @@ func cloneRemote(ctx context.Context, srcDB *doltdb.DoltDB, remoteName, branch s
 		for _, brnch := range branches {
 			branch = brnch.GetPath()
 
-			if branch == "master" {
+			if branch == doltdb.MasterBranch {
 				break
 			}
 		}
 	}
 
+	// If we couldn't find a branch but the repo cloned successfully, it's empty. Initialize it instead of pulling from
+	// the remote.
+	// This would be unnecessary if we properly initialized the storage for a repository when we created it on dolthub.
+	// If we do that, this code can be removed.
+	performPull := true
+	if branch == "" {
+		name := dEnv.Config.GetStringOrDefault(env.UserNameKey, "")
+		email := dEnv.Config.GetStringOrDefault(env.UserEmailKey, "")
+
+		if *name == "" {
+			return errhand.BuildDError(fmt.Sprintf("error: could not determine user name. run dolt config --global --add %[1]s", env.UserNameKey)).Build()
+		} else if *email == "" {
+			return errhand.BuildDError("error: could not determine email. run dolt config. run dolt config --global --add %[1]s", env.UserEmailKey).Build()
+		}
+
+		err = dEnv.InitDBWithTime(ctx, types.Format_Default, *name, *email, doltdb.CommitNowFunc())
+		if err != nil {
+			return errhand.BuildDError("error: could not initialize repository").AddCause(err).Build()
+		}
+
+		branch = doltdb.MasterBranch
+		performPull = false
+	}
+
```

Comment:
Maybe put this all in a function initIfEmpty or something like that with your comment at the top.

Reply:
Done.